core/shared/platform/zephyr/zephyr_file.c

```diff
@@ -711,22 +713,24 @@ os_renameat(os_file_handle old_handle, const char *old_path,
     char abs_old_path[MAX_FILE_NAME + 1];
     char abs_new_path[MAX_FILE_NAME + 1];
 
+    GET_FILE_SYSTEM_DESCRIPTOR(old_handle->fd, ptr);
+
+    char *path = strdup(new_path);
+    if (path == NULL) {
+        return __WASI_ENOMEM;
+    }
+
     snprintf(abs_old_path, MAX_FILE_NAME, "%s/%s", prestat_dir, old_path);
     snprintf(abs_new_path, MAX_FILE_NAME, "%s/%s", prestat_dir, new_path);
 
     int rc = fs_rename(abs_old_path, abs_new_path);
     if (rc < 0) {
+        free(path);
         return convert_errno(-rc);
     }
 
-    GET_FILE_SYSTEM_DESCRIPTOR(old_handle->fd, ptr);
-
-    ptr->path = strdup(new_path);
-    if (ptr->path == NULL) {
-        ptr->path = old_path;
-        return __WASI_ENOMEM;
-    }
-
+    free(ptr->path);
```

Comment:
just to be sure, IIUC, the core idea of this patch is to `free(ptr->path)` before assigning a new value. Is that correct?